File: PerformanceAnalysis/src/main/java/com/android/performanceanalysis/data/HomeData.java
package com.android.performanceanalysis.data;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by jcy on 2018/8/8.
 */

public class HomeData {
    public static List<ItemView> addDevTotalRes = new ArrayList<>();

    static {
        addDeviceItem("", "性能分析与优化");
        addDeviceItem("&#xe620;", "WebView");
        addDeviceItem("&#xe620;", "AOP 学习");
        addDeviceItem("&#xe620;", "App启动优化");
        addDeviceItem("&#xe620;", "App内存优化");
        addDeviceItem("&#xe620;", "App布局优化");
        addDeviceItem("&#xe620;", "App卡顿优化");
        addDeviceItem("&#xe620;", "App线程优化");
        addDeviceItem("&#xe620;", "App网络优化");
        addDeviceItem("&#xe620;", "App电量优化");
        addDeviceItem("&#xe620;", "App瘦身优化");
        addDeviceItem("&#xe620;", "App稳定性优化");
        addDeviceItem("&#xe620;", "IdleHandler的使用");
    }

    private static void addDeviceItem(String icon, String desc) {
        addDevTotalRes.add(new ItemView(icon, desc));
    }

    public static class ItemView {
        public String icon;
        public String desc;

        public ItemView(String icon, String desc) {
            this.icon = icon;
            this.desc = desc;
        }
    }
}

File: Lib-StudySet/Architecture/src/main/java/com/android/architecture/data/HomeData.kt
package com.android.architecture.data

import java.util.ArrayList

/**
 * Created by jcy on 2018/8/8.
 */

object HomeData {
    var addDevTotalRes: MutableList<ItemView> = ArrayList()

    init {
        addDeviceItem("", "架构组件 demo")
        addDeviceItem("&#xe620;", "Jetpack(数据绑定部分)之Lifecycles")
        addDeviceItem("&#xe620;", "Jetpack(数据绑定部分)之ViewModel")
        addDeviceItem("&#xe620;", "Jetpack(数据绑定部分)之LiveData")
        addDeviceItem("&#xe620;", "Jetpack(数据绑定部分)之DataBinding")
        addDeviceItem("&#xe620;", "Jetpack(数据绑定部分)之Paging")
        addDeviceItem("&#xe620;", "Jetpack(数据绑定部分)之Room")
        addDeviceItem("&#xe620;", "Jetpack(数据绑定部分)之Navigation")
        addDeviceItem("&#xe620;", "Jetpack(数据绑定部分)之WorkManager")
    }

    private fun addDeviceItem(icon: String, desc: String) {
        addDevTotalRes.add(ItemView(icon, desc))
    }

    class ItemView(var icon: String, var desc: String)
}
